src/ring_theory/roots_of_unity.lean

```diff
@@ -715,4 +716,79 @@ lemma nth_roots_one_eq_bind_primitive_roots {ζ : R} {n : ℕ} (hpos : 0 < n)
 
 end integral_domain
 
+section minimal_polynomial
+
+open minimal_polynomial
+
+variables {n : ℕ} {K : Type*} [field K] {μ : K} (h : is_primitive_root μ n) (hpos : 0 < n)
+
+include n μ h hpos
+
+/--`μ` is integral over `ℤ`. -/
+lemma is_integral : is_integral ℤ μ :=
+begin
+  use (X ^ n - 1),
+  split,
+  { exact (monic_X_pow_sub_C 1 (ne_of_lt hpos).symm) },
+  simp only [((is_primitive_root.iff_def μ n).mp h).left, eval₂_one, eval₂_X_pow, eval₂_sub,
+  sub_self],
+end
+
+/--The minimal polynomial of `μ` is monic. -/
+lemma is_monic_minimal_polynomial : (minimal_polynomial (is_integral h hpos)).monic :=
+(minimal_polynomial.monic (is_integral h hpos))
+
+/--The minimal polynomial of `μ` is primitive. -/
+lemma is_primitive_minimal_polynomial : is_primitive
+(minimal_polynomial (is_integral h hpos)) := monic.is_primitive
+(is_monic_minimal_polynomial h hpos)
+
+variables [char_zero K]
+
+/--`μ ` is integral over `ℚ`. -/
+lemma is_algebraic : _root_.is_integral ℚ μ :=
+@is_integral_of_is_scalar_tower ℤ ℚ K _ _ _ (algebra_int ℚ) _ _ add_comm_group.int_is_scalar_tower
+  μ (is_integral h hpos)
+
+/--The minimal polynomial of `μ` over `ℚ` is the same as the minimal polynomial over `ℤ`. -/
+lemma min_poly_rational_eq_min_poly_integer : minimal_polynomial (is_algebraic h hpos)
+   = map (int.cast_ring_hom ℚ) (minimal_polynomial (is_integral h hpos)) :=
+begin
+  refine (unique' (is_algebraic h hpos) _ _ _).symm,
+  { exact (is_primitive.int.irreducible_iff_irreducible_map_cast (is_primitive_minimal_polynomial h
+  hpos)).1 (irreducible (is_integral h hpos)) },
+  { have htower := is_scalar_tower.aeval_apply ℤ ℚ K μ (minimal_polynomial
+    (is_integral h hpos)),
+    rw [minimal_polynomial.aeval] at htower,
+    exact htower.symm },
+  { exact monic_map _ (is_monic_minimal_polynomial h hpos) }
+end
+
+/--If `P : polynomial ℤ` is primitive and has `μ` as root, then the minimal polynomial of `μ`
+over `ℤ` divides `P`. -/
+lemma minimal_polynomial_dvd {P : polynomial ℤ} (hprim : is_primitive P)
```

Comment:
Is there some generalisation lurking in the background? I think minpolys always divide primitive polys. Is that right?